File: OKGIP/backend/src/main/java/com/okip/service/hr/HRService.java
package com.okip.service.hr;

import java.util.List;
import java.util.Map;

import com.okip.dto.hr.EmployeeApprovalResponseDTO;
import com.okip.dto.hr.HRDashboardDTO;
import com.okip.dto.hr.PendingEmployeeDTO;

public interface HRService {

    EmployeeApprovalResponseDTO approveEmployee(Long employeeId);

    EmployeeApprovalResponseDTO rejectEmployee(Long employeeId);

    List<PendingEmployeeDTO> getPendingEmployees();

    HRDashboardDTO getDashboard();

    List<Map<String, Object>> getEmployees();
    List<Map<String, Object>> getDepartments();
    List<Map<String, Object>> getJobRoles();
    List<Map<String, Object>> getWorkforceSkills();
    List<Map<String, Object>> getCompetencies();
    List<Map<String, Object>> getKnowledgeGaps();
    Map<String, Object> getTrainingAnalytics();
    List<Map<String, Object>> getAssessments();
}

File: OKGIP/frontend/src/services/hrService.ts
import API from "@/api/axios";

export interface PendingEmployee { employeeId:number; employeeCode:string; firstName:string; lastName:string; officialEmail:string; }
export interface HRDashboardData {
  totalEmployees:number; pendingApprovals:number; employeesWithSkillGaps:number; criticalSkillGaps:number;
  employeesInTraining:number; trainingCompletionRate:number; averageLearningProgress:number; averageAssessmentScore:number; averageSkillImprovement:number;
  activeMentorships:number; totalSkills:number; totalSkillAssignments:number;
  departmentSummaries:Array<{departmentName:string;employeeCount:number;averageProficiency:number;averageGapPercentage:number}>;
  topSkillGaps:Array<{skillName:string;affectedEmployees:number;averageGapPercentage:number;severity:string}>;
  trainingStatus:Array<{status:string;count:number;percentage:number}>;
  assessmentSummary:Array<{assessmentType:string;attempts:number;averagePercentage:number}>;
  employeeGrowth:Array<{month:string;count:number}>;
}
export type HRRow = Record<string, any>;

export const hrService = {
  getDashboard: async () => (await API.get<HRDashboardData>("/hr/dashboard")).data,
  getPendingEmployees: async () => (await API.get<PendingEmployee[]>("/hr/pending")).data,
  approveEmployee: async (id:number) => (await API.put(`/hr/approve/${id}`)).data,
  rejectEmployee: async (id:number) => (await API.put(`/hr/reject/${id}`)).data,
  getEmployees: async () => (await API.get<HRRow[]>("/hr/employees")).data,
  getDepartments: async () => (await API.get<HRRow[]>("/hr/departments")).data,
  getJobRoles: async () => (await API.get<HRRow[]>("/hr/job-roles")).data,
  getWorkforceSkills: async () => (await API.get<HRRow[]>("/hr/workforce-skills")).data,
  getCompetencies: async () => (await API.get<HRRow[]>("/hr/competencies")).data,
  getKnowledgeGaps: async () => (await API.get<HRRow[]>("/hr/knowledge-gaps")).data,
  getTrainingAnalytics: async () => (await API.get<HRRow>("/hr/training-analytics")).data,
  getAssessments: async () => (await API.get<HRRow[]>("/hr/assessments")).data,
};
export default hrService;
